docs/templates/feature.md

```diff
@@ -2,42 +2,41 @@
 
 *Features* are the most granular entity in the [feature store](feature_store.md) and are
 logically grouped by [feature groups](feature_group.md).
+Features in the same feature groups are semantically related to the feature groups primary keys.
 
 The storage location of a single feature is determined by the
 [feature group](feature_group.md). Hence, enabling a [feature group](feature_group.md)
 for online storage will make a feature available as an online feature.
 
-New features can be appended to [feature groups](feature_group.md), however, to drop
-features, a new [feature group](feature_group.md) version has to be created. When
-appending features it is possible to specify a default value which is used for existing
-feature vectors in the [feature group](feature_group.md) for the new feature.
+## Features Taxonomy
 
-{{feature}}
+Within a [feature group](feature_group.md) there are different categories of features:
+
+* **Primary Keys**: These features describe the entity contained in a feature group (e.g. customer features, user features, ...). Within a feature group there could be more than one feature describing the primary key.
```

Comment:
A primary key is not a feature. It is not used to train a model. It is only used for online feature store for retrieving features. I prefer to use the term 'column' for that reason.

The columns that uniquely identify an entity in a feature group (e.g., the customer, the transaction, ...). A primary key may consist of multiple columns, that is, a composite primary key.

Reply:
I changed it, though I don't fully agree. While technically correct that you don't use primary keys as features to train the model, introducing the word "column" in the documentation I think just creates confusion for new users - those reading the documentation in the first place. 
We also don't have different APIs to define what's a feature and what's a primary key, same thing in the UI, everything is referred as feature - mostly for simplicity and ease of use.

So in a sense, by changing the word "feature" with the word "column" we are trading simplicity for correctness.

Reply:
Primary keys, event_time, and features are all columns. 
Maybe we need to define that earlier in the docs. We have to differentiate them - it's not simpler to confuse them. I agree we should minimize using the word column, but if we have to refer to the collection of Primary keys, event_time, and features - i don't see what other word to use.

I would also note another important terminology issue. The primary key for RonDB is not the same as Hudi. In RonDB, it doesn't include event_time, but it does in Hudi. For this reason, it the book, i refer to the PK as the "entity key(s)". I call it a primary key when we get to HSFS. See more in the book, see section 1.1.1.